### PR TITLE
Remove string.__index error

### PR DESCRIPTION
**Pull request by GitSparTV**
We don't need this exclusive error.
Let it be `attempt to call method foo` instead of `attempt to index a string value with bad key ('foo' is not part of the string library)`.

### CODE REVIEW

**Comment by SneakySquid** [comment]
thanks spar very good :+1:

**Comment by SneakySquid** [comment]
thanks gs very good 👍

File: garrysmod/lua/includes/extensions/string.lua [hunk 1]
```diff
@@ -289,12 +289,10 @@ local meta = getmetatable( "" )
 
 function meta:__index( key )
 	local val = string[ key ]
-	if ( val ) then
+	if ( val ~= nil ) then
```
Comment (dvdvideo1234) [inline garrysmod/lua/includes/extensions/string.lua:292]: Is it more accurate when `key` is nil to raise error ? This way `("test")[nil]` will do nothing.

Comment (Kefta) [reply garrysmod/lua/includes/extensions/string.lua:292]: All this change does is allow
```Lua
string.foo = false
print(string.foo)
```
to not error.
